aii-core/src/main/perl/DHCP.pm

```diff
@@ -495,43 +495,56 @@ sub read_input
     return $error;
 }
 
+# update the dhcp config file and restart daemon
+sub update_and_restart 
+{
+    my $self = shift;
+
+    my $filename = $self->option('dhcpconf');
+    $self->debug(1, "AII::DHCP: going to update dhcpd configuration $filename if needed");
+    my ($error, $changed) = $self->update_dhcp_config($filename);
+    if ($error) {
+        $self->error("AII::DHCP: failed to update dhcpd configuration $filename");
+        return;
+    }
+
+    # restart dhcpd daemon
+    if ($changed) {
+        if ($self->option('norestart')) {
+            $self->verbose("AII::DHCP: config changed but norestart set!");
+        } else {
+            my $cmd = $self->option('restartcmd');
+            $self->debug(1, "AII::DHCP: restarting dhcpd daemon using cmd '$cmd'");
+            # expects an arayref
+            $self->restart_daemon([split(/\s+/, $cmd)]);
+        } 
+    } else {
+        $self->verbose("AII::DHCP: no changes to $filename: daemon not restarted");
+    }
+    return 1;
+}
+
+# return true if dhcp config need changes
+sub nodes_to_change {
+    my $self = shift;
+    return  return (@{$self->{NTC}} + @{$self->{NTR}}) ? 1 : 0;
+}
 
 # return 1 on failure, 0 on success
 sub configure
 {
     my $self = shift;
 
     # process command line options
-    $self->debug(1, "aii-dhcp: reading cmd line or input files");
+    $self->debug(1, "AII::DHCP: reading cmd line or input files");
     if ($self->read_input()) {
-        $self->error("aii-dhcp: failed to process cmd line or input files");
+        $self->error("AII::DHCP: failed to process cmd line or input files");
         return(1);
     }
 
     # update dhcpd configuration file
-    if( scalar(@{$self->{NTC}}) > 0 || scalar(@{$self->{NTR}}) > 0 ) {
-        my $filename = $self->option('dhcpconf');
-
-        $self->debug(1, "aii-dhcp: going to update dhcpd configuration $filename");
-        my ($error, $changed) = $self->update_dhcp_config($filename);
-        if ($error) {
-            $self->error("aii-dhcp: failed to update dhcpd configuration $filename");
-            return(1);
-        }
-
-        # restart dhcpd daemon
-        if ($self->option('norestart')) {
-            $self->verbose("aii-dhcp: dhcpd daemon do not restart (norestart set)");
-        } else {
-            if ($changed) {
-                my $cmd = $self->option('restartcmd');
-                $self->debug(1, "aii-dhcp: restarting dhcpd daemon using cmd '$cmd'");
-                # expects an arayref
-                $self->restart_daemon([split(/\s+/, $cmd)]);
-            } else {
-                $self->verbose("aii-dhcp: no changes to $filename: daemon not restarted");
-            }
-        }
+    if ($self->nodes_to_change()) {
+		return 1 if (!$self->update_and_restart()) ;
     } else {
         $self->debug(1, "aii-nbp: there are no changes to dhcpd configuration to make");
```

Comment:
`aii-nbp`? also this is confusing, message should read something like `no nodes to configure and this no chagnes to dhcp configuration`